packages/mobile/src/localCurrency/saga.ts

```diff
@@ -57,10 +58,17 @@ function* updateLocalCurrencyRate() {
 
   while (true) {
     try {
-      yield put(fetchCurrentRate())
+      const lastSuccessfulUpdate: number | undefined = yield select(
+        (state: RootState) => state.localCurrency.lastSuccessfulUpdate
+      )
+      if (lastSuccessfulUpdate && Date.now() - lastSuccessfulUpdate < MIN_UPDATE_INTERVAL) {
+        yield take()
```

Comment:
what does this do? Does this just take any action that's dispatched?

Reply:
Yes.

Reply:
What this does is check the date difference on every state change and only trigger the update after 24 hours.

Reply:
That seems wasteful, why check again of EVERY state change? 
Can we just use the approach we talked about the other day of only checking once per app session?

Reply:
It might feel wasteful, but this is not different from what happens with `connect` on the component side. Each `mapStateToProps` is called on every state change.

I wanted this to be resilient to long running sessions (app in background and MainActivity remaining active).
I think a good alternative approach would be to use [AppState](https://facebook.github.io/react-native/docs/appstate) and trigger this time difference check when the app state becomes `active`.
What do you think?

Reply:
This still feels like overkill to support a 1% case. I don't think I prefer the AppState approach either, as I don't feel the complexity is worth the benefit.
I _would_ be open to having this run when a user triggers a refresh (e.g. pulling down on wallet home).
https://github.com/celo-org/celo-monorepo/blob/master/packages/mobile/src/home/saga.ts#L36

I believe the auto-refresh will have this run relatively often which should meet the requirement you're going for here

Reply:
It's not a 1% case. I believe it will happen to most users. 

Not a big fan of now coupling this with the balance auto refresh either. 😸 
It uses a 10 secs timer loop. This could be seen as wasteful too when the app is not in use.
We might as well do the same in that module if we go that way.

Reply:
once every 10 seconds seems better than every state change right? 
I'm still imagining you'd keep the `lastSuccessfulUpdate` check.
It seems like we're going around a bit on this, let's chat about it tmrw morning :)